DRAFT: Raise PhpStan to level 8

nezaniel opened the pull request:
**Upgrade instructions**

Adjust all dependent code to potential type changes

**Review instructions**

Please check whether all
* types have been applied correctly
* all fallback values make sense

**Checklist**

- [ ] Code follows the PSR-12 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

Commit:
WIP: Adjust Neos.Utility.Unicode

Commit:
Adjust Neos.Utility.Schema

Commit:
Adjust Neos.Utility.Pdo

Commit:
Adjust Neos.Utility.ObjectHandling

Commit:
Adjust Neos.Utility.MediaTypes

Commit:
Adjust Neos.Utility.Files

Commit:
Adjust Neos.Utility.Arrays

Commit:
Adjust Neos.Kickstarter

Commit:
Adjust Neos.Http.Factories

Commit:
Adjust Neos.Http.Factories

Commit:
Adjust Neos.FluidAdaptor

Commit:
Adjust Neos.Flow.Log

Commit:
Adjust Neos.Flow Validation subcontext

Commit:
Adjust Neos.Flow Utility subcontext

Commit:
Adjust Neos.Flow SignalSlot subcontext

Commit:
Adjust Neos.Flow Session subcontext

Commit:
Adjust Neos.Flow Security subcontext

Commit:
Adjust Neos.Flow ResourceManagement subcontext

Commit:
Adjust Neos.Flow Reflection subcontext

Commit:
Adjust Neos.Flow Property subcontext

Commit:
Adjust Neos.Flow Persistence subcontext

Commit:
Adjust Neos.Flow Package subcontext

Commit:
Adjust Neos.Flow ObjectManagement subcontext

Commit:
Adjust Neos.Flow Mvc subcontext

Commit:
Adjust Neos.Flow Monitor subcontext

Commit:
Adjust Neos.Flow Log subcontext

Commit:
Adjust Neos.Flow I18n subcontext

Commit:
Adjust Neos.Flow Http subcontext

Commit:
Adjust Neos.Http.Factories to Neos.Flow Http subcontext adjustments

Commit:
Adjust Neos.Flow Error subcontext

Commit:
Adjust Neos.Flow core

Commit:
Adjust Neos.Flow Configuration subcontext

Commit:
Adjust Neos.Flow Composer subcontext

Commit:
Adjust Neos.Flow Cli subcontext

Commit:
Adjust Neos.Flow Cache subcontext

Commit:
Adjust Neos.Flow Aop subcontext

Commit:
Adjust Neos.Flow Annotations subcontext

Commit:
Adjust Neos.Flow application layer

Commit:
Adjust Neos.Error.Messages

Commit:
Adjust Neos.Eel

Commit:
Adjust Neos.Cache

Commit:
add scanDirectories folders for special classes

mhsdesign commented on Neos.Cache/Classes/Backend/AbstractBackend.php:
```diff
@@ -50,28 +50,25 @@ abstract class AbstractBackend implements BackendInterface
      *
-     * @param EnvironmentConfiguration $environmentConfiguration
-     * @param array $options Configuration options - depends on the actual backend
+     * @param ?EnvironmentConfiguration $environmentConfiguration @todo is this ever null and does that even make sense?
```
fyi solved this in an followup pr https://github.com/neos/flow-development-collection/pull/3470

Commit:
Relax class type checks

Commit:
Remove further class checks

Commit:
remove var dumps

mhsdesign committed:
PATCH: Make PHP 8.2 compatible

mhsdesign committed:
PATCH: Exception during the reflection of the `JoinPoint`

> Cannot declare class Neos\Flow\Aop\Exception, because the name is already in use

`$exception` in line 64 with the doc type changed to `?\Exception` from `\Exception` is at fault. It seems the expansion from `?\Exception` behaves differently than `\Exception|null` and thus another `class_exist` checks leads to multiple autoloading.

nezaniel commented:
Closed in favor of #3474
